upf/pfcp.c

```diff
@@ -794,12 +794,15 @@ decode_f_teid (u8 * data, u16 length, void *p)
 {
   pfcp_f_teid_t *v = p;
 
-  if (length < 5)
-    return PFCP_CAUSE_INVALID_LENGTH;
+//  if (length < 5)
+//   return PFCP_CAUSE_INVALID_LENGTH;
 
   v->flags = get_u8 (data) & 0x0f;
-  v->teid = get_u32 (data);
-  length -= 5;
+  if (!(v->flags & F_TEID_CH))
+  {
+    v->teid = get_u32 (data);
+    length -= 5;
+  }
```

Comment:
* remove the flag element length before the if, decrement should be 4 and check the length inside the if block before reading the value
* does not follow VPP coding style, should be indented 2 space more
* should go into the block in line 829/830
* IPv4/v6 address block (lines 832-848) needs to be moved up in CH flag check (line 829/830)
* CHID block (lines 850-856) needs to be moved up into CH flag check (line 814/815).

```suggestion
  length--;

  if (!(v->flags & F_TEID_CH))
    {
      if (length < 4)
        return PFCP_CAUSE_INVALID_LENGTH;
      v->teid = get_u32 (data);
      length -= 4;
    }
```

Reply:
I've refactored decoder part